exp/zapslog/slog.go

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2023 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+//go:build zapslog
+
+package zapslog
+
+import (
+	"context"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"golang.org/x/exp/slog"
+)
+
+type handler struct {
+	logger *zap.Logger
+}
+
+var _ slog.Handler = (*handler)(nil)
+
+// group holds all the Attrs saved in a slog.GroupValue.
+type group struct {
+	attrs []slog.Attr
+}
+
+func (g *group) MarshalLogObject(enc zapcore.ObjectEncoder) error {
+	for _, attr := range g.attrs {
```

Comment:
Two things:

- We don't need a struct here. Just a `type group []slog.Attr` should be enough.
- 'group' is too general a name for package-level scope. We should qualify it to something more specific that is less likely to collide with a variable or package name elsewhere. `groupObject` may be better.

```suggestion
// groupObject holds all the Attrs saved in a slog.GroupValue.
type groupObject []slog.Attr

func (gs groupObject) MarshalLogObject(enc zapcore.ObjectEncoder) error {
	for _, attr := range gs {
```